src/test/unit_tests/jest_tests/test_config_dir_restructure_upgrade_script.test.js

```diff
@@ -0,0 +1,654 @@
+        const account_names_obj = {};
+        await P.map_with_concurrency(5000, account_ids, async account_id => {
+            const new_account_data = { _id: String(account_id), name: 'old_account' + account_id, user: 'root' };
+            await write_manual_old_account_config_file(default_config_fs, new_account_data);
+            account_names_obj[new_account_data.name] = new_account_data;
+        });
+        await move_old_accounts_dir(default_config_fs, Object.keys(account_names_obj), mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir(account_names_obj);
+    }, TEST_TIMEOUT);
+
+    it('move_old_accounts_dir() - .backup_accounts_dir already exists and contains some of the accounts in accounts/ dir', async () => {
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        await create_fresh_path(default_config_fs.identities_dir_path);
+        await create_fresh_path(default_config_fs.accounts_by_name_dir_path);
+
+        const account_ids = Array.from({ length: 200 }, (_, i) => Number(i + 1));
+        const account_names_obj = {};
+        const symlink_options = { symlink_name: false, symlink_access_key: false };
+        await P.map_with_concurrency(100, account_ids, async account_id => {
+            const new_account_data = { _id: String(account_id), name: 'old_account' + account_id, user: 'root' };
+            await write_manual_old_account_config_file(default_config_fs, new_account_data, symlink_options);
+            account_names_obj[new_account_data.name] = new_account_data;
+            if (account_id % 2 === 0) {
+                await write_manual_config_file(CONFIG_TYPES.ACCOUNT, default_config_fs, new_account_data, undefined, symlink_options);
+            }
+        });
+        await move_old_accounts_dir(default_config_fs, Object.keys(account_names_obj), mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir(account_names_obj);
+    });
+});
+
+describe('create_account_access_keys_index_if_missing', () => {
+    beforeAll(async () => {
+        await fail_test_if_default_config_dir_exists('test_config_dir_restructure', default_config_fs);
+    });
+
+    beforeEach(async () => {
+        await create_config_dir(config.NSFS_NC_DEFAULT_CONF_DIR);
+        await default_config_fs.create_config_dirs_if_missing();
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        await create_fresh_path(hidden_old_accounts_path);
+    }, TEST_TIMEOUT);
+
+    afterEach(async () => {
+        await clean_config_dir(default_config_fs);
+    }, TEST_TIMEOUT);
+
+
+    it('create_account_access_keys_index_if_missing() - single regular account', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
```

Comment:
I think it might be confusing, as the mock has only access_key_id (without the secret_key).